OpenRA.Mods.Common/Traits/Mobile.cs

```diff
@@ -428,29 +132,11 @@ public void SetLocation(CPos from, SubCell fromSub, CPos to, SubCell toSub)
 			ToSubCell = toSub;
 			AddInfluence();
 
-			// Tunnel condition is added/removed when starting the transition between layers
-			if (toCell.Layer == CustomMovementLayerType.Tunnel && conditionManager != null &&
-					!string.IsNullOrEmpty(Info.TunnelCondition) && tunnelToken == ConditionManager.InvalidConditionToken)
-				tunnelToken = conditionManager.GrantCondition(self, Info.TunnelCondition);
-			else if (toCell.Layer != CustomMovementLayerType.Tunnel && tunnelToken != ConditionManager.InvalidConditionToken)
-				tunnelToken = conditionManager.RevokeCondition(self, tunnelToken);
-
-			// Play submerging animation as soon as it starts to submerge (before applying the condition)
-			if (toCell.Layer == CustomMovementLayerType.Subterranean && fromCell.Layer != CustomMovementLayerType.Subterranean)
-			{
-				if (!string.IsNullOrEmpty(Info.SubterraneanTransitionSequence))
-					self.World.AddFrameEndTask(w => w.Add(new SpriteEffect(self.World.Map.CenterOfCell(fromCell), self.World, Info.SubterraneanTransitionImage,
-						Info.SubterraneanTransitionSequence, Info.SubterraneanTransitionPalette)));
-
-				if (!string.IsNullOrEmpty(Info.SubterraneanTransitionSound))
-					Game.Sound.Play(SoundType.World, Info.SubterraneanTransitionSound);
-			}
-
-			// Grant the jumpjet condition as soon as the actor starts leaving the ground layer
-			// The condition is revoked from FinishedMoving
-			if (toCell.Layer == CustomMovementLayerType.Jumpjet && conditionManager != null &&
-					!string.IsNullOrEmpty(Info.JumpjetCondition) && jumpjetToken == ConditionManager.InvalidConditionToken)
-				jumpjetToken = conditionManager.GrantCondition(self, Info.JumpjetCondition);
+			// Tunnel condition is added/removed when starting the transition between layers.
+			// Jumpjet condition is granted at this point, too.
```

Comment:
Are these comments still appropriate (or needed)? I thought we kind of wanted to make this more abstract.